### PR TITLE
VIH-7650 retry failed bookings

### PR DESCRIPTION
**Pull request by shaed-parkar**
### JIRA link (if applicable) ###

VIH-7650

### Change description ###

* display retry confirmation booking for failed confirmations, as long as the scheduled time is in the future

image

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```
Previously it was not possible to retry a failed booking

### CODE REVIEW

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts [hunk 1]
```diff
@@ -126,6 +128,14 @@ export class BookingDetailsComponent implements OnInit, OnDestroy {
         }
     }
 
+    get canRetryConfirmation(): boolean {
+        if (!this.booking || this.booking.status !== BookingStatus.Failed) {
+            return false;
+        }
+        const scheduledTime = moment(this.booking.scheduled_date_time);
+        return scheduledTime.isAfter(moment(new Date()));
```
Comment (robertbennett1998) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:136]: What would happen if the page loads with a few seconds left and then extends over the time then the button was pressed. Should confirm hearing also call this?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts [hunk 2]
```diff
@@ -221,6 +231,7 @@ export class BookingDetailsComponent implements OnInit, OnDestroy {
         this.$subscriptions.push(
             this.videoHearingService.getHearingById(this.hearingId).subscribe(
                 newData => {
+                    newData.status = BookingStatus.Created;
```
Comment (robertbennett1998) [inline AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:234]: What is this doing?

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts (hunk 1 above)
Comment (shaed-parkar) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:136]: the button is bound to the property so if the time passes, like confirm, the button would disappear from the UI

File: AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts (hunk 2 above)
Comment (shaed-parkar) [reply AdminWebsite/AdminWebsite/ClientApp/src/app/bookings-list/booking-details/booking-details.component.ts:234]: ah, hangover from manual testing. will remove that

**Review by robertbennett1998 — APPROVED** [review]
LGTM

**Pull request merged**